### DISCUSSION TITLE
Jaeger TLS with ElasticSearch - Helm Chart

### DISCUSSION THREAD

**Discussion by n1vgabay** [jaegertracing/jaeger#3442] [Q&A] [answered]
Hi all,

Using:
- k8s 1.21
- eks cluster with elasticsearch service deployed apart of other services
- Jaeger helm chart
- Create TLS from elasticsearch via job and assign pem in secret, other services in cluster communicate without problem using the pem from secret.

Goals:
- Use this secret who contains the elastic pem private key to let jaeger communicate with elastic securly.

From Jaeger docs, I've got a bit confused and would love to get some help.

Do I have to create 2 config maps for it? (prefer not to and just assign)

It's a small tweak that will let jaeger works, already deployed to cluster but collector and query can't work without TLS.

**Comment by n1vgabay** [comment]
Someone please??

**Comment by pavolloffay** [comment]
A better place for this issue would be jaeger helm charts repository. 

The TLS in jaeger is configured via flags. For more details see https://www.jaegertracing.io/docs/1.29/cli/#jaeger-all-in-one-elasticsearch. Now you need to figure out how the helm chart configures flags. 

Note that the spark-dependencies job does not support TLS yet.

**Comment by n1vgabay** [comment, marked as answer]
Ok, so the ticket can be closed.

Not sure why jaeger is using configmap (I'd prefer something different and more secure but it is what it is..)

So I created init-jaeger and did all the manipulation I had to do and worked like a charm.
